tests/test_main.py

```diff
@@ -87,15 +88,17 @@ def pkg_with_dashes(site_dir):
         metadata = metadata_dir / 'METADATA'
         with metadata.open('w', encoding='utf-8') as strm:
             strm.write('Version: 1.0\n')
-        return 'my-pkg'
+        return 'my-pkg', 'my_pkg'
 
     def test_dashes_in_dist_name_found_as_underscores(self):
         """
         For a package with a dash in the name, the dist-info metadata
         uses underscores in the name. Ensure the metadata loads.
         """
-        pkg_name = self.pkg_with_dashes(self.site_dir)
-        assert version(pkg_name) == '1.0'
+        pkg_name, norm_pkg_name = self.pkg_with_dashes(self.site_dir)
+        dist = distribution(pkg_name)
+        assert dist._normalized_name == norm_pkg_name
```

Comment:
Instead of testing the internal implementation details, I'd rather have a test that captures the failed expectation - namely that entry points of two packages whose file system names vary only by normalization won't deduplicate.